apps/tutorial-servo-can-control/src/TutorialServo.cpp

```diff
@@ -0,0 +1,37 @@
+#include "TutorialServo.h"
+
+TutorialServo::TutorialServo(PinName servoPin, float servoRangeInDegrees, float minPulsewidthInMs, float maxPulsewidthInMs)
+              : m_servoRangeInDegrees(servoRangeInDegrees), m_minPulsewidthInMs(minPulsewidthInMs), m_maxPulsewidthInMs(maxPulsewidthInMs)
+{
+    // Initialize servo pin and period
+    m_servoPwmOut(servoPin);
+    m_servoPwmOut.period_ms(m_periodInMs);
+
+}
+
+// Set servo position (ex. 45deg)
+void TutorialServo::setPositionInDegrees(const float degrees){
+    float finalPulsewidth = ((degrees/m_servoRangeInDegrees) * (m_maxPulsewidthInMs-m_minPulsewidthInMs)) + m_minPulsewidthInMs;
```

Comment:
what if someone sends a value outside the range

Reply:
I think I would just add a check to see if the value was inside the range, if so run the code as normal. If not, return some flag/error code that alerts an invalid value was passed.

Reply:
yeah thatd be a fair thing to do. but lets say you are forced to use the api shown in the training doc. that is this function must return void. try and think of a way to fullfill the instruction as best you can without hurting the motor

Reply:
Oh ok, i guess one way would be to wrap around the degrees (depending on what the range). So for example, if the range was 180 degrees, passing in 200 degrees would just wrap around to 20 degrees and passing in -50 deg would wrap around to 130 degrees.

Reply:
or i should say, any negative values, you'd just take the absolute values to make it simple

Reply:
Implementation wise off the top of my head, I'd replace (degrees/m_servoRangeInDegrees) (fabs(degrees) / m_servoRangeInDegrees) - floor(degrees/m_servoRangeInDegrees)